### PR TITLE
CC-461 combobox keyboard handling

### PR DESCRIPTION
**Pull request by tjheffner**
## Summary

- Adds appropriate keyboard handling to the combobox per https://app.zenhub.com/workspaces/contention-classification-640a24db5dcd08457cebbc79/issues/gh/department-of-veterans-affairs/vagov-claim-classification/461

- adds handling for the Tab, Enter, Escape, Spacebar, Up & Down Arrow Keys. 

## Related issue(s)
 https://app.zenhub.com/workspaces/contention-classification-640a24db5dcd08457cebbc79/issues/gh/department-of-veterans-affairs/vagov-claim-classification/461

## Testing done

- Verified that keyboard handling worked as described for each key.

- Enter text into box.
- Up and down arrow keys should change the highlighted element.
- Hit 'Enter'. The highlighted element should be selected, options minimized & focus on the text input.
- Continue typing, the autosuggest should be highlighted. Hit Escape, the options should minimize and updated text should remain the same, focus on the text input.
- Type some more, include a space. The input should take a space no problem. When a different item is selected (via arrow key navigation), space should function the same as Enter (highlighted option selected, options minimized, focus on the text input)
- Type some more, hit `Tab`. The options should minimize, updated text remain the same, `update` or `save` button should have focus.

## What areas of the site does it impact?

526ez form, revised disabilities page

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### CODE REVIEW

File: src/applications/disability-benefits/all-claims/components/ComboBox.jsx [hunk 1]
```diff
@@ -99,9 +203,19 @@ export class ComboBox extends React.Component {
       filteredOptions: [],
       highlightedIndex: 0,
     });
-    this.inputRef.current.focus();
     const { onChange } = this.props;
     onChange(option);
+    // Send focus to input element for additional user input.
+    this.inputRef.current.focus();
```
Comment (tyler-spangler6) [inline src/applications/disability-benefits/all-claims/components/ComboBox.jsx:209]: When the user selects an option from the dropdown with the mouse, the focus should move back to the input field. When i was testing this locally, the focus was not moving back to the input field from the list. I played around with the below to set focus and it seemed to work but just wanted to share. 
```
const vaTextFieldShadowRoot = this.inputRef.current.shadowRoot
const inputField = vaTextFieldShadowRoot.querySelector('input')
inputField.focus()
```

**Review by tyler-spangler6 — COMMENTED** [review]
I was not sure where to put the below in the code but there are two tests in the ticket that I could not find in the code:
- If the dropdown list is open and the user clicks on an area outside the dropdown the list should close and keep the text that the user input with the focus in the input box
- The condition should autopopulate in the input field when the user clicks edit (this might be more appropriate in a separate ticket if it is a harder problem to solve)

**Comment by tjheffner** [comment]
All feedback has been addressed save for this one:

> The condition should autopopulate in the input field when the user clicks edit (this might be more appropriate in a separate ticket if it is a harder problem to solve)

I think this should be a separate ticket, as I believe it isn't related to the combobox specifically, but the interactions with the schema + array field

**Review by tyler-spangler6 — APPROVED** [review]

**Review by lukey-luke — APPROVED** [review]
LGTM

**Pull request merged**